src/rand.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Authors (alpha order): Adrian Catangiu @acatangiu, Alexandra Lordache @aghecenco, Alexandru Agache @alexandruag, Diana Popa @dianpopa, Matt Forrester @forbesmyester, Radu Weiss @raduweiss
+
+extern crate libc;
+use std::ffi::OsString;
+use std::str;
+
+pub fn timestamp_cycles() -> u64 {
+    #[cfg(target_arch = "x86_64")]
+    // Safe because there's nothing that can go wrong with this call.
+    unsafe {
+        std::arch::x86_64::_rdtsc() as u64
```

Comment:
The return value of rdtsc() is some sort of predictable, so it should not be used for cryptographic.

Reply:
Agree, I've added comments saying as such and also changed function names to ensure people understand it's a pseudo random, not the real thing.